pkg/client/client.go

```diff
@@ -75,12 +76,15 @@ func NewSyncClient(configFile, logFile, recordsFile string, routineNum, retries
 func (c *Client) Run() {
 	fmt.Println("Start to generate sync tasks, please wait ...")
 
-	var finishChan = make(chan int, c.routineNum)
+	//var finishChan = make(chan struct{}, c.routineNum)
 
 	// opem num of goroutines and wait c for close
 	openRoutinesGenTaskAndWaitForFinish := func() {
+		wg := sync2.WaitGroup{}
 		for i := 0; i < c.routineNum; i++ {
+			wg.Add(1)
```

Comment:
How about wg.Add(c.routineNum) for only once before the loop?

Reply:
wg.Add(c.routineNum) is ok, but i think this may be more easy to understand.

Reply:
> How about wg.Add(c.routineNum) for only once before the loop?

wg.Add(c.routineNum) is ok, but i think this may be more easy to understand.

Reply:
Fine, it does seem better.